### PR TITLE
Update samples to Dropbox v2 API now that v1 API has been discontinued

### PR DESCRIPTION
**Pull request by mitchyboy9**
Hello,

I wanted to use NoNonesense FilePicker in my project to select directories from Dropbox, but it wouldn't work because Dropbox has discontinued the v1 API used by the samples.

I have re-written the Dropbox sample to use the v2 API and thought I would share it for others to benefit.

The only thing that I'm not sure about, and would appreciate your help with, is the change I made in commit a0349ef, which I did to remove a NullPointerException when I first tried to run the Dropbox sample.  I feel like I have missed something and that recyclerView in DropboxFilePickerFragment.java should somehow get automatically assigned.  Is there a better way that I could have used to do this?

### CODE REVIEW

File: sample/src/main/java/com/nononsenseapps/filepicker/sample/dropbox/DropboxFilePickerFragment.java [hunk 1]
```diff
@@ -57,6 +58,7 @@ protected View inflateRootView(LayoutInflater inflater, ViewGroup container) {
         View view = inflater.inflate(R.layout.fragment_loading_filepicker, container, false);
         // And bind the progress bar
         progressBar = (ProgressBar) view.findViewById(R.id.progressBar);
+        recyclerView = (RecyclerView) view.findViewById(android.R.id.list);
```
Comment (spacecowboy) [inline sample/src/main/java/com/nononsenseapps/filepicker/sample/dropbox/DropboxFilePickerFragment.java]: Delete this line.

File: sample/src/main/java/com/nononsenseapps/filepicker/sample/dropbox/DropboxFilePickerFragment.java [hunk 2]
```diff
@@ -16,39 +16,40 @@
...
 @SuppressLint("ValidFragment")
-public class DropboxFilePickerFragment
-        extends AbstractFilePickerFragment<DropboxAPI.Entry> {
-
-    private final DropboxAPI<AndroidAuthSession> dbApi;
+public class DropboxFilePickerFragment extends AbstractFilePickerFragment<Metadata> {
+    private static final String TAG = "DbxFilePickerFragment";
+    private final DbxClientV2 dropboxClient;
     private ProgressBar progressBar;
     private RecyclerView recyclerView;
```
Comment (spacecowboy) [inline sample/src/main/java/com/nononsenseapps/filepicker/sample/dropbox/DropboxFilePickerFragment.java]: Delete this line.

**Comment by spacecowboy** [comment]
And if it's not too much trouble, would you mind adding an entry for `build-tools-26.0.2` somewhere here: https://github.com/spacecowboy/NoNonsense-FilePicker/blob/master/.travis.yml#L10 ? (I could do this seperately if you prefer otherwise)

**Comment by mitchyboy9** [comment]
Thanks very much for your help with the RecyclerView issue.  It's working fine for me now as well.

I've updated the Travis build tools (and the Example and Sample modules, so everything is at the same level).

Finally, I made a tweak to the Dropbox login code because, if you didn't authenticate with Dropbox when taken to the app/website to authorise the sample, you would get stuck in a loop where the sample would try to authenticate again...and again until you either killed the app, or did, actually, log in to Dropbox.

**Comment by spacecowboy** [comment]
Excellent. Thanks so much.

**Pull request merged**